app/Platform/Actions/LoadGameWithRelationsAction.php

```diff
@@ -65,13 +68,89 @@ public function execute(Game $game, AchievementFlag $flag, ?GameAchievementSet $
+     * 4. (if all the same points) The first achievement in the list.
+     */
+    private function computeGroupRepresentativeBadges(Game $game): void
+    {
+        foreach ($game->gameAchievementSets as $gameAchievementSet) {
+            $achievementSet = $gameAchievementSet->achievementSet;
+            $achievements = $achievementSet->achievements;
+
+            if ($achievements->isEmpty()) {
+                continue;
+            }
+
+            // Group achievements by their group ID.
+            $achievementsByGroup = $achievements->groupBy(fn (Achievement $a) => $a->pivot->achievement_group_id ?? -1);
+
+            foreach ($achievementSet->achievementGroups as $group) {
+                $groupAchievements = $achievementsByGroup->get($group->id, collect());
+                $representative = $this->findRepresentativeAchievement($groupAchievements);
+                $group->representative_badge_url = $representative?->badge_unlocked_url;
+            }
+
+            // Also compute the representative badge for ungrouped achievements.
+            $ungroupedAchievements = $achievementsByGroup->get(-1, collect());
+            $ungroupedRepresentative = $this->findRepresentativeAchievement($ungroupedAchievements);
+            $achievementSet->ungrouped_badge_url = $ungroupedRepresentative?->badge_unlocked_url;
+        }
+    }
+
+    /**
+     * Find the representative achievement for a group based on priority rules.
+     *
+     * @param Collection<int, Achievement> $achievements
+     */
+    private function findRepresentativeAchievement(Collection $achievements): ?Achievement
+    {
+        if ($achievements->isEmpty()) {
+            return null;
+        }
+
+        $sortedAchievements = $achievements->sortBy(fn (Achievement $a) => $a->pivot->order_column ?? $a->DisplayOrder);
+
+        // Priority 1: Last win condition achievement.
+        $winConditions = $sortedAchievements->filter(fn (Achievement $a) => $a->type === AchievementType::WinCondition);
+        if ($winConditions->isNotEmpty()) {
+            return $winConditions->last();
+        }
+
+        // Priority 2: Last progression achievement.
+        $progressions = $sortedAchievements->filter(fn (Achievement $a) => $a->type === AchievementType::Progression);
+        if ($progressions->isNotEmpty()) {
+            return $progressions->last();
+        }
+
+        // Priority 3: Highest value achievement (by points).
+        $maxPoints = $sortedAchievements->max(fn (Achievement $a) => $a->points);
+        $highestPointAchievements = $sortedAchievements->filter(fn (Achievement $a) => $a->points === $maxPoints);
+
+        if ($highestPointAchievements->count() === 1) {
+            return $highestPointAchievements->first();
+        }
+
+        // Priority 4: If they're all same points (or multiple with the highest), return the first achievement in the list.
+        // This is very likely an edge case.
```

Comment:
I would suggest using the last achievement in the group as the fallback - assuming the group is sorted in the order the player would earn them, the last achievement would be the "completion condition" for the group.

Although that might be adequately covered by higest value achievement?

And this may not be as uncommon as you'd like. If we allow things to be group into "base game", "post game", and "DLC", neither "post game" nor "DLC" would have any progression markings.

Reply:
Updated in latest so priority 4 is now the last achievement in the group.